debug/src/debug.js

```diff
@@ -56,6 +65,44 @@ export function initDebug() {
 			throw new Error('Invalid type passed to createElement(): '+(Array.isArray(type) ? 'array' : type));
 		}
 
+		if ((type==='thead' || type==='tfoot' || type==='thead') && parentType!=='table') {
+			console.error(
+				'Improper nesting of table.' +
+				'Your <thead/tbody/tfoot> should have a <table> parent.'
+				+ serializeVNode(vnode)
+			);
+		}
+		else if (type==='tr' && (parentType!=='thead' && parentType!=='tfoot' && parentType!=='tbody')) {
```

Comment:
According to the W3 HTML validator this is a valid table:

```html
<table>
  <tr>
    <td>foo</td>
  </tr>
</table>
```

The [spec](https://html.spec.whatwg.org/multipage/tables.html#the-table-element) has this to say about the children of a `<table>` node:

> In this order: optionally a `caption` element, followed by zero or more `colgroup` elements, followed optionally by a `thead` element, followed by either zero or more `tbody` elements or one or more `tr` elements, followed optionally by a `tfoot` element, optionally intermixed with one or more script-supporting elements.

It looks like we just need to check the parent type for `table` too :+1: